### PR TITLE
refactor: remove discord.js enums and use discord-api-types enums instead

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

TODO:

- [x] Make sure spots weren't missed
- [x] Test for any regressions

Note: I won't be updating the enums for any components. I'll wait for #7040 to be merged before I modify anything with components.

**Status and versioning classification:**
Please move lines that apply to you out of the comment:
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -153,3 +153,24 @@ exports.WelcomeChannel = require('./structures/WelcomeChannel');
 exports.WelcomeScreen = require('./structures/WelcomeScreen');
 
 exports.WebSocket = require('./WebSocket');
+
+// External
+exports.ActivityType = require('discord-api-types').ActivityType;
```
Comment (NotSugden) [inline src/index.js:158]: feel like it would be worth it to use a variable instead of repeated `require`s 😄

Comment (iCrawl) [reply src/index.js:158]: Probably not because it will most likely fuck with bundlers again, it's why we reverted all the dynamic imports in general.

Comment (vladfrangu) [reply src/index.js:158]: Would it really fuck bundlers up if we did `const v9 = require('discord-api-types/v9');` and export from there?

Comment (iCrawl) [reply src/index.js:158]: Most likely increase bundle size for no apparent reason because of tree-shaking

Comment (vladfrangu) [inline src/index.js:158]: Ohhhh my gawd. Please move to versioned imports 🙏

File: src/managers/ApplicationCommandPermissionsManager.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 'use strict';
 
 const { Collection } = require('@discordjs/collection');
+const { ApplicationCommandPermissionType } = require('discord-api-types');
 const BaseManager = require('./BaseManager');
 const { Error, TypeError } = require('../errors');
-const { ApplicationCommandPermissionTypes, APIErrors } = require('../util/Constants');
+const { APIErrors } = require('../util/Constants');
```
Comment (vladfrangu) [inline src/managers/ApplicationCommandPermissionsManager.js:7]: Isn't this also exported in types?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/structures/Interaction.js [hunk 3]
```diff
@@ -141,55 +141,55 @@ class Interaction extends Base {
...
+    return InteractionType[this.type] === InteractionType.ApplicationCommand;
   }
 
   /**
    * Indicates whether this interaction is a {@link ChatInputCommandInteraction}.
    * @returns {boolean}
    */
   isChatInputCommand() {
-    return InteractionTypes[this.type] === InteractionTypes.APPLICATION_COMMAND && typeof this.targetId === 'undefined';
+    return InteractionType[this.type] === InteractionType.APPLICATION_COMMAND && typeof this.targetId === 'undefined';
```
Comment (vladfrangu) [inline packages/discord.js/src/structures/Interaction.js:152]:
```suggestion
    return InteractionType[this.type] === InteractionType.ApplicationCommand && typeof this.targetId === 'undefined';
```

File: packages/discord.js/src/structures/interfaces/TextBasedChannel.js [hunk 4]
```diff
@@ -358,3 +357,4 @@ module.exports = TextBasedChannel;
 
 // Fixes Circular
 const MessageManager = require('../../managers/MessageManager');
+const { InteractionType } = require('discord-api-types/v9');
```
Comment (vladfrangu) [inline packages/discord.js/src/structures/interfaces/TextBasedChannel.js:360]: Why's it down here?

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/client/actions/InteractionCreate.js [hunk 5]
```diff
@@ -16,18 +17,18 @@ class InteractionCreateAction extends Action {
     // Resolve and cache partial channels for Interaction#channel getter
     this.getChannel(data);
 
-    let InteractionType;
+    let _InteractionType;
```
Comment (kyranet) [inline packages/discord.js/src/client/actions/InteractionCreate.js:20]: Alternatively, can we call this `InteractionStructure` or `InteractionClass`?

File: packages/discord.js/src/structures/GuildAuditLogs.js [hunk 6]
```diff
@@ -111,56 +112,53 @@ const Targets = {
...
-  STICKER_UPDATE: 91,
-  STICKER_DELETE: 92,
-  GUILD_SCHEDULED_EVENT_CREATE: 100,
-  GUILD_SCHEDULED_EVENT_UPDATE: 101,
-  GUILD_SCHEDULED_EVENT_DELETE: 102,
-  THREAD_CREATE: 110,
-  THREAD_UPDATE: 111,
-  THREAD_DELETE: 112,
-};
+// const Actions = {
```
Comment (kyranet) [inline packages/discord.js/src/structures/GuildAuditLogs.js:115]: Any reason why this is commented?

**Review by kyranet — CHANGES_REQUESTED** [review]
This PR makes me very happy, we're finally getting PascalCased enums!

Just a couple of things :P

File: packages/discord.js/src/structures/GuildAuditLogs.js (hunk 6 above)
Comment (suneettipirneni) [reply packages/discord.js/src/structures/GuildAuditLogs.js:115]: Forgot to remove this in favor for `AuditLogEvent` in -types

**Review by iCrawl — APPROVED** [review]

**Comment by ImRodry** [comment]
The API documents pretty much every type as UPPER_CASE ([example](https://discord.com/developers/docs/resources/channel#channel-object-channel-types)), why are we switching to PascalCase?

**Review by thebigwolfy — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by almeidx — CHANGES_REQUESTED** [review]
These were not updated:

https://github.com/suneettipirneni/discord.js/blob/84f0da43041887c13897ba40f70f9b2e2179f277/packages/discord.js/src/structures/MessageMentions.js#L4

https://github.com/suneettipirneni/discord.js/blob/84f0da43041887c13897ba40f70f9b2e2179f277/packages/discord.js/src/structures/ContextMenuCommandInteraction.js#L5

https://github.com/suneettipirneni/discord.js/blob/84f0da43041887c13897ba40f70f9b2e2179f277/packages/discord.js/src/structures/CommandInteraction.js#L7

https://github.com/suneettipirneni/discord.js/blob/84f0da43041887c13897ba40f70f9b2e2179f277/packages/discord.js/src/structures/BaseMessageComponent.js#L4

https://github.com/suneettipirneni/discord.js/blob/84f0da43041887c13897ba40f70f9b2e2179f277/packages/discord.js/src/structures/MessageActionRow.js#L4

https://github.com/suneettipirneni/discord.js/blob/84f0da43041887c13897ba40f70f9b2e2179f277/packages/discord.js/src/structures/MessageButton.js#L5

https://github.com/suneettipirneni/discord.js/blob/84f0da43041887c13897ba40f70f9b2e2179f277/packages/discord.js/src/structures/MessageComponentInteraction.js#L6

https://github.com/suneettipirneni/discord.js/blob/84f0da43041887c13897ba40f70f9b2e2179f277/packages/discord.js/src/structures/MessageSelectMenu.js#L4

https://github.com/suneettipirneni/discord.js/blob/84f0da43041887c13897ba40f70f9b2e2179f277/packages/discord.js/src/managers/GuildScheduledEventManager.js#L7

https://github.com/suneettipirneni/discord.js/blob/84f0da43041887c13897ba40f70f9b2e2179f277/packages/discord.js/src/structures/GuildScheduledEvent.js#L9

https://github.com/suneettipirneni/discord.js/blob/84f0da43041887c13897ba40f70f9b2e2179f277/packages/discord.js/src/managers/GuildScheduledEventManager.js#L7

***

This was removed from the constants. So it should be removed from here too. However, it's being used in other files (the doc type) and it doesn't exist anymore

https://github.com/suneettipirneni/discord.js/blob/84f0da43041887c13897ba40f70f9b2e2179f277/packages/discord.js/src/util/Constants.js#L585

***

This type should be removed:

https://github.com/suneettipirneni/discord.js/blob/84f0da43041887c13897ba40f70f9b2e2179f277/packages/discord.js/typings/index.d.ts#L2766

**Comment by suneettipirneni** [comment]
> These were not updated:

I fixed all of these except for the components, as stated in the PR description those will updated in another PR.

**Pull request merged**